### PR TITLE
Relax parsing of labels

### PR DESCRIPTION
**Pull request by purkhusid**
**What type of PR is this?**

 Bug fix

**What package or component does this PR mostly affect?**

Gazelle extension using the `label` package

**What does this PR do? Why is it needed?**

The parsing in the `labels` package is too strict. This PR updates the regex to allow more characters that Bazel is OK with.

My exact use case was that I was generating TypeScript targets for a NextJS application which has has the following folder structure:
```
pages/
    entity/
       [entityId]/
           index.ts
```

This caused issues with target-determinator(https://github.com/bazel-contrib/target-determinator) which uses the labels package to parse affected labels.

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This configuration file was automatically generated by Gitpod.
+# Please adjust to your needs (see https://www.gitpod.io/docs/introduction/learn-gitpod/gitpod-yaml)
+# and commit this file to your remote git repository to share the goodness with others.
```
Comment (fmeum) [inline .gitpod.yml:3]: Please drop this file from the PR

File: label/label.go [hunk 2]
```diff
@@ -63,8 +63,8 @@ var NoLabel = Label{}
 
 var (
 	labelRepoRegexp = regexp.MustCompile(`^@$|^[A-Za-z.-][A-Za-z0-9_.-]*$`)
-	labelPkgRegexp  = regexp.MustCompile(`^[A-Za-z0-9/._@-]*$`)
-	// This was taken from https://docs.bazel.build/versions/main/build-ref.html#name
+	labelPkgRegexp  = regexp.MustCompile(`^[A-Za-z0-9 !\\\"#$%&'()*+,\-./;<=>?[\]^_\x60{|}~'@]*$`)
+	// This was taken from https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/cmdline/LabelValidator.java
 	labelNameRegexp = regexp.MustCompile("^[A-Za-z0-9!%-@^_` \"#$&'()*-+,;<=>?\\[\\]{|}~/.]*$")
```
Comment (fmeum) [inline label/label.go:68]: Is this also out of date? Except for colon and backslash, essentially all printable chars appear to be supported.

File: label/label.go [hunk 3]
```diff
@@ -63,8 +63,8 @@ var NoLabel = Label{}
 
 var (
 	labelRepoRegexp = regexp.MustCompile(`^@$|^[A-Za-z.-][A-Za-z0-9_.-]*$`)
-	labelPkgRegexp  = regexp.MustCompile(`^[A-Za-z0-9/._@-]*$`)
-	// This was taken from https://docs.bazel.build/versions/main/build-ref.html#name
+	labelPkgRegexp  = regexp.MustCompile(`^[A-Za-z0-9 !\\\"#$%&'()*+,\-./;<=>?[\]^_\x60{|}~'@]*$`)
+	// This was taken from https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/cmdline/LabelValidator.java
```
Comment (fmeum) [inline label/label.go:67]: Could you extend this comment a bit to explain how you derived the regex from the code at that link?

File: .gitpod.yml (hunk 1 above)
Comment (purkhusid) [reply .gitpod.yml:3]: Ah sorry, will do.

File: label/label.go (hunk 2 above)
Comment (purkhusid) [reply label/label.go]: Yup, updated it as well

File: label/label.go (hunk 3 above)
Comment (purkhusid) [reply label/label.go:66]: Added a bit more details to the comment

File: label/label.go [hunk 4]
```diff
@@ -63,9 +63,15 @@ var NoLabel = Label{}
...
-	// This was taken from https://docs.bazel.build/versions/main/build-ref.html#name
-	labelNameRegexp = regexp.MustCompile("^[A-Za-z0-9!%-@^_` \"#$&'()*-+,;<=>?\\[\\]{|}~/.]*$")
+	// This was taken from https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/cmdline/LabelValidator.java
+	// Package names may contain all 7-bit ASCII characters except:
+	// 0-31 (control characters)
+	// 58 ':' (colon) - target name separator
+	// 92 '\' (backslash) - directory separator (on Windows); may be allowed in the future
+  	// 127 (delete)
+	// Target names may contain the same characters
+	labelPkgRegexp  = regexp.MustCompile(`^[A-Za-z0-9 !\"#$%&'()*+,\-./;<=>?[\]^_\x60{|}~'@]*$`)
```
Comment (fmeum) [inline label/label.go:73]: This regex would match the comment more obviously: `^[\x20-\x39\x3B-\x5B\x5D-\x7E]*$`.

File: label/label.go [hunk 5]
```diff
@@ -63,9 +63,15 @@ var NoLabel = Label{}
...
+	labelNameRegexp = regexp.MustCompile(`^[A-Za-z0-9 !\"#$%&'()*+,\-./;<=>?[\]^_\x60{|}~'@]*$`)
```
[the remark this answers is not in the document]
Comment (purkhusid) [reply label/label.go]: Fixed

File: label/label.go (hunk 4 above)
Comment (purkhusid) [reply label/label.go]: Nice, learned something new today! Fixed

**Review by fmeum — APPROVED** [review]
@tyler-french Could you test at Uber?

**Comment by purkhusid** [comment]
Any update on this?

**Comment by purkhusid** [comment]
Conflicts resolved

**Review by fmeum — APPROVED** [review]

**Pull request merged**